packages/dashboard/src/components/TilePicker/index.js

```diff
@@ -1,48 +1,113 @@
+} from './styled';
+import useRegisterChildren from './hooks/useRegisterChildren';
 
+const TilePickerContext = createContext();
 /**
  * TilePicker can be either radio buttons or checkboxes that are visually enhanced.
  */
 export default function TilePicker(props) {
     const { style, id, children, value, multiple, className, name, onChange } = props;
 
-    const nameUnique = useUniqueIdentifier('tile-picker');
-    const groupNameId = name || nameUnique;
+    const containerRef = useRef();
+    const scrollableRef = useRef();
+    const [width, setWidth] = useState();
+    const [current, setCurrent] = useState(0);
 
-    const handleChange = (optionName, isChecked) => {
-        if (multiple) {
-            if (Array.isArray(value)) {
-                const currentValue = isChecked
-                    ? [...value, optionName]
-                    : value.filter((item) => item !== optionName);
+    useEffect(() => {
+        setWidth(containerRef.current.getClientRects()[0].width);
+    }, []);
 
-                return onChange(currentValue);
-            }
+    useWindowResize(() => {
+        setWidth(containerRef.current.getClientRects()[0].width);
+    });
+
+    const isCarousel = width && children.length > Math.floor(width / 200);
+    const delta = isCarousel ? width && Math.floor((width - 32) / 200) : children.length;
+    const isDisablePrevious = current < 1;
+    const isDisableNext = current + delta >= children.length;
+
+    const nameUnique = useUniqueIdentifier('tile-picker');
+    const groupName = name || nameUnique;
 
-            const currentValue = isChecked ? [optionName] : [];
-            return onChange(currentValue);
+    const { activeChildren, registerChild, unregisterChild } = useRegisterChildren({
+        containerRef,
+    });
+
+    useEffect(() => {
+        if (width) {
+            scrollableRef.current.scrollLeft = current * 200;
         }
-        return onChange(optionName);
-    };
+    }, [current, width]);
 
-    const context = {
-        groupName: groupNameId,
-        onChange: handleChange,
-        value,
-        multiple,
-    };
+    const setFocused = useCallback(
+        (childRef) => {
+            const activeIndex = activeChildren.current.findIndex((child) => child.ref === childRef);
+            if (activeIndex > -1) {
+                if (activeIndex < current) {
+                    setCurrent(activeIndex);
+                }
+                if (activeIndex >= current + delta) {
+                    setCurrent(activeIndex - delta + 1);
+                }
+            }
+        },
+        [activeChildren, current, delta],
+    );
+
+    const context = { registerChild, unregisterChild, setFocused };
 
     return (
-        <StyledContainer id={id} className={className} style={style}>
-            <Provider value={context}>{children}</Provider>
+        <StyledContainer id={id} className={className} style={style} ref={containerRef}>
+            <RenderIf isTrue={!!width}>
```

Comment:
not use double negate here, we let pass isTrue any value then the coercion is done already internally